addon/addon-test-support/@ember/test-helpers/setup-context.ts

```diff
@@ -98,7 +98,21 @@ export function isTestContext(context: BaseContext): context is TestContext {
   );
 }
 
-let __test_context__: BaseContext | undefined;
+/**
+  @private
+  @param {Object} it the global object to test
+  @returns {Boolean} it exists
+*/
+function check(it: any) {
```

Comment:
why have this check?

why not:
```js
const globalObject = 
  (typeof globalThis === 'object' && globalThis) ||
  (typeof window === 'object' && window) ||
  (typeof self === 'object' && self) ||
  (typeof global === 'object' && global);
```

Reply:
first because it's heavily tested code from core-js implementation
second it tests for actual global instead of accidental variables